### PR TITLE
algorithm: kosaraju

### PR DESCRIPTION
**Pull request by Adrito-M**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1215"></a>

### CODE REVIEW

**Comment by Adrito-M** [comment]
@raklaptudirm @appgurueu This PR solves issue #1214

File: package-lock.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
         "standard": "^17.0.0"
       },
       "engines": {
-        "node": ">=18"
+        "node": ">=16.6.0"
```
Comment (appgurueu) [inline package-lock.json:26]: Please get rid of this change. Looks fine from a first glance otherwise.

Comment (Adrito-M) [reply package-lock.json]: Made the requested changes

**Comment by Adrito-M** [comment]
Please Take A Look

**Review by appgurueu — DISMISSED** [review]
Consider hiding the class as an implementation detail by only exporting a function `kosaraju(graph)` (which uses the class).

**Review by appgurueu — APPROVED** [review]

**Comment by Adrito-M** [comment]
@raklaptudirm please take a look

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**